src/Deposit.js

```diff
@@ -1099,4 +1103,174 @@ export default class Deposit {
+
+  /**
+   * Get the current collateralization level for this Deposit.
+   * Collateralisation will be 0% if the deposit is in liquidation.
+   * @return {number} Percentage collateralisation, as an integer. eg. 149%
+   */
+  async getCollateralisationPercentage() {
+    return parseInt(
+      await this.contract.methods.getCollateralizationPercentage().call()
+    )
+  }
+
+  /**
+   * Get the initial collateralization level for this Deposit.
+   * @return {number} Percentage collateralisation, as an integer. eg. 150%
+   */
+  async getInitialCollateralizedPercentage() {
+    return parseInt(
+      await this.contract.methods.getInitialCollateralizedPercent().call()
+    )
+  }
+
+  /**
+   * Get the first threshold for deposit undercollateralisation.
+   * If the collateralization level falls below this percentage, the Deposit can
+   * get courtesy-called.
+   * The deposit can be courtesy called using `Deposit.notifyCourtesyCall`.
+   * @return {number} Percentage collateralisation, as an integer. eg. 125%
+   */
+  async getUndercollateralizedThresholdPercent() {
+    return parseInt(
+      await this.contract.methods
+        .getUndercollateralizedThresholdPercent()
+        .call()
+    )
+  }
+
+  /**
+   * Get the threshold for severe deposit undercollateralisation.
+   * If the collateralization level falls below this percentage, the Deposit
+   * can be liquidated.
+   * Liquidation can be initiated using `Deposit.notifyUndercollateralizedLiquidation`.
+   * @return {number} Percentage collateralisation, as an integer. eg. 110%
+   */
+  async getSeverelyUndercollateralizedThresholdPercent() {
+    return parseInt(
+      await this.contract.methods
+        .getSeverelyUndercollateralizedThresholdPercent()
+        .call()
+    )
+  }
+
+  /**
+   * Notify the contract that the signers are undercollateralized,
+   * and move the deposit into a pre-liquidation state.
+   */
+  async notifyCourtesyCall() {
+    await EthereumHelpers.sendSafely(this.contract.methods.notifyCourtesyCall())
+  }
+
+  /**
+   * Exit the courtesy call state.
+   * Only callable if the deposit is sufficiently collateralised.
+   */
+  async exitCourtesyCall() {
+    await EthereumHelpers.sendSafely(this.contract.methods.exitCourtesyCall())
+  }
+
+  /**
+   * Notify the contract that the deposit is severely undercollateralised,
+   * and begin liquidation of the signer bonds.
+   *
+   * The bonds are auctioned in a falling-price auction. The value of
+   * the bonds can be queried using `Deposit.auctionValue`, and bids placed
+   * using `Deposit.purchaseSignerBondsAtAuction`.
+   */
+  async notifyUndercollateralizedLiquidation() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.notifyUndercollateralizedLiquidation()
+    )
+  }
+
+  /**
+   * Purchases the signer bonds and closes the liquidation auction.
+   *
+   * Note: The caller must approve the Deposit contract to spend
+   * `lotSize` amount of TBTC.
```

Comment:
This feels like it should be built into this particular function. If people want to call the underlying contract function independently of the “blessed path”, they can and should call `.contract` and interact with its methods directly.